src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -347,7 +347,7 @@ public function reset()
         }
 
         foreach ($this->multi->openHandles as [$ch]) {
-            if (\is_resource($ch)) {
+            if (\is_resource($ch) || $ch instanceof \CurlHandle) {
```

Comment:
Just wondering, why are these is_resource checks necessary in the first place? Are they getting closed somehow without being removed?

Reply:
Good question indeed. I haven't found a reason why `$ch` should be anything else than a cURL handle. And the tests still pass if I remove the two guard clauses.

Maybe @nicolas-grekas knows more about why we need(ed?) this check.

Reply:
I've seen strange things happen at destructing/shutdown time, this might be related.